plugins/wacom-i2c/fu-wac-device.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2018 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#ifndef __FU_WAC_DEVICE_H
+#define __FU_WAC_DEVICE_H
+
+#include "fu-udev-device.h"
+#include "dfu-element.h"
+
+G_BEGIN_DECLS
+
+#define FU_TYPE_WAC_DEVICE (fu_wac_device_get_type ())
+G_DECLARE_DERIVABLE_TYPE (FuWacDevice, fu_wac_device, FU, WAC_DEVICE, FuUdevDevice)
+
+struct _FuWacDeviceClass
+{
+	FuUdevDeviceClass	parent_class;
+	gboolean		 (*write_firmware)	(FuDevice	*self,
+							 GPtrArray	*chunks,
+							 GError		**error);
+};
+
+typedef struct hid_descriptor {
+	guint16 wHIDDescLength;
+	guint16 bcdVersion;
+	guint16 wReportDescLength;
+	guint16 wReportDescRegister;
+	guint16 wInputRegister;
+	guint16 wMaxInputLength;
+	guint16 wOutputRegister;
+	guint16 wMaxOutputLength;
+	guint16 wCommandRegister;
+	guint16 wDataRegister;
+	guint16 wVendorID;
+	guint16 wProductID;
+	guint16 wVersion;
+	guint16 RESERVED_HIGH;
+	guint16 RESERVED_LOW;
+} FuWacHidDescriptor;
```

Comment:
Aren't these standard hid descriptor properties?

Reply:
Maybe, yes. Do we have a Linux header somewhere with these in?

Reply:
Well, we do export the version/vendor/product through the standard HID device (so hidraw has access to it). As for exporting the rest (the command registers), I really do not see the point.

IIRC, there was a systemd bug filed against udev to also export the HID vendor/product/version in udev in case of i2c and bluetooth devices (because right now only USB are populating the ID_MODEL_ID and ID_VENDOR_ID)

Reply:
I don't think we need this at all if we switch to hidraw.